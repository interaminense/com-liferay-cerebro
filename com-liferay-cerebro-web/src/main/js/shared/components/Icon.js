import Component, {Config} from 'metal-jsx';
import getCN from 'classnames';

import analyticsConstants from '../util/constants';

const SIZES = ['sm', 'md', 'lg', 'xl', 'xxl', 'xxxl'];

class Icon extends Component {
	render() {
		const {color, monospaced, size, symbol} = this.props;

		const classes = getCN(
			'icon-root',
			'lexicon-icon',
			`lexicon-icon-${symbol}`,
			{
				[`${color}-color`]: color,
				['icon-monospaced']: monospaced,
				[`icon-size-${size}`]: size
			}
		);

		return (
			<svg {...this.otherProps()} class={classes}>
				<use
					xlink:href={`${
						analyticsConstants.pathThemeImages
					}/lexicon/icons.svg#${symbol}`}
				/>
			</svg>
		);
	}
}

Icon.PROPS = {
	color: Config.string(),
	elementClasses: Config.string(),
	monospaced: Config.bool(),
	size: Config.oneOf(SIZES),
	symbol: Config.string().required()
};

Icon.SIZES = SIZES;

export default Icon;