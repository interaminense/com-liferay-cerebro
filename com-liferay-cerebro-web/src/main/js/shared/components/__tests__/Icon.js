import Icon from '../Icon';

describe('Icon', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new Icon({
			symbol: 'file'
		});

		expect(component).toMatchSnapshot();
	});
});