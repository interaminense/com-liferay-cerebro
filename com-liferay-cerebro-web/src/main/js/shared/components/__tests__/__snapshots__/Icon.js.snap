// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render 1`] = `
<svg class="icon-root lexicon-icon lexicon-icon-file">
  <use xlink:href="/images/lexicon/icons.svg#file"></use>
</svg>
`;
